### PR TITLE
19276 - Fixed price renderer issue

### PR DESCRIPTION
**Pull request by sarfarazbheda**
### Fixed Issues (if relevant)

1. magento/magento2#19276: Fixed price renderer issue

### Manual testing scenarios (*)

1. Select Any category, select Swatches (color and size) from category page. Price will reflect on the current product after selecting color and size.(i.e. Swatch)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19376) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @sarfarazbheda. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js
Comment (orlangur) [inline app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js]: Isn't `$product.find` enough? Please put common part into variable or rewrite as
```
$product.find(...)[condition ? 'show' : 'hide']();
```

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js
Comment (sarfarazbheda) [reply app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js]: Hello,

I got your point, I will make changes as per your suggestions

File: app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js
Comment (orlangur) [inline app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js]: Teh line is too long now, I would suggest a variable like

```
isShow = typeof result != 'undefined' && result.oldPrice.amount !== result.finalPrice.amount;
```

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js
Comment (sarfarazbheda) [reply app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js]: Applied changes and pushed again.

Comment (orlangur) [reply app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js]: @sarfarazbheda please do not change unneded lines and remove brackets around `(isShow)`.

After that squash changes into a single commit.

Comment (sarfarazbheda) [reply app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js]: yes, Made changes and pushed branch again

Comment (orlangur) [reply app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js]: @sarfarazbheda I still see unneeded lines of code changed. No lines besides 922-926 should be changed, like it was before.

Comment (sarfarazbheda) [reply app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js]: Reverted my old changes and committed code

**Comment by orlangur** [comment]
@sarfarazbheda thanks, changes look good now, please squash them into a single commit, force push and we are good to go 👍

**Comment by sarfarazbheda** [comment]
As per your comment above, pushed changed into single commit. Let me know once you merge that branch. Thanks.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3608](https://jira.corp.magento.com/browse/ENGCOM-3608) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@sarfarazbheda thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by sdzhepa** [comment]
✔️ QA passed

Was tested by steps described in the initial issue https://github.com/magento/magento2/issues/19276

1. Created swatches
2. Created several Configurable products with the same swatch color
3. Created Catalog Price Rule to apply a discount (special price)
4. Applied changes to the final_price.phtml template file as mentioned in https://magento.stackexchange.com/questions/196090/magento-2-display-old-price-and-special-price-in-category-list
 Replace
```
<?php if (!$block->isProductList() && $block->hasSpecialPrice()): ?>
<span class="old-price sly-old-price no-display">
```
with
```
<?php if ($block->hasSpecialPrice()): ?>
<span class="old-price sly-old-price">
```
5. Flush caches and reindex
6. Go to Storefront on Catgory page

**Result**
pr19376

**Comment by meker12** [comment]
Draft release note: When you apply a price rule to a configurable product with swatches, Magento now shows the special price only for the configurable product that has the swatch attribute specified in the price rule. Previously, Magento displayed the special price for all configurable products that had the matching swatch attribute.

**Pull request merged**
